### ISSUE TITLE
Compatibility with griffe 0.15

### ISSUE THREAD

**Issue by martibosch** [mkdocstrings/mkdocstrings#416]
**Describe the bug**
When building the documentation, i.e., `mkdocs build` with griffe 0.15, I get a ` KeyError: <class 'ast.Sub'>`, whereas the same build succeeds with griffe 0.14 (holding the same versions for all the other libraries of the environment).

**To Reproduce**
The build logs provide all the information required to reproduce the issue (as well as the error traceback here: https://github.com/martibosch/agrometeo-geopy/runs/5953852082?check_suite_focus=true#step:6:206). You can see an example of a successful build with griffe 0.14 here: https://github.com/martibosch/agrometeo-geopy/runs/5674152832?check_suite_focus=true.

**Expected behavior**
The command `mkdocs build` should successfully build the documentation.

**Information (please complete the following information):**
- OS: ubuntu 20.04
- Browser: irrelevant
- `mkdocstrings` version: 0.18.1

Thank you in advance.

**Comment by pawamoy** [comment]
Hi, thanks for the report! Version 0.15 implements some features for the Numpy parser. In this particular case, the issue is that Griffe tries to parse your docstring annotation into an expression, and it chokes on a `Sub` node, which indicates there's a `-` in your annotation. I've looked at your code and found this example: `region : str, list-like, geopandas.GeoSeries, geopandas.GeoDataFrame, geometric`.

I'll catch such errors and discard them, as users might not want to benefit from that feature and write their annotations like text, not Python code (typing unions, etc.). In the meantime, and only if you find this beneficial, you can rewrite this kind of annotations as typing annotations, with `typing.Union` or future annotations like `str | typing.Sequence | geopandas.GeoSeries | geopandas.GeoDataFrame | geometric`.

**Comment by pawamoy** [comment]
```yaml
# mkdocs.yml

plugins:
- mkdocstrings
    handlers:
      python:
        import:
        - https://geopandas.org/en/stable/objects.inv
```


https://user-images.githubusercontent.com/3999221/162569493-d7a9a5ca-0b8b-4ee8-abdc-257350ca22a1.mp4

**Comment by pawamoy** [comment]
That should be fixed in version 0.16.0.

**Comment by martibosch** [comment]
Hello @pawamoy!

thank you for addressing this so quickly. I will consider moving towards typing or future annotations, but in the meantime, thank you again for your fix.

Best,
Martí
